index.js

```diff
@@ -1,56 +1,156 @@
+}
+
+
+/* Patch $.ajax to support expired CSRF tokens */
+function addRetrySupport(retryURL, parseResponse) {
+    const originalAjax = $.ajax;
+
+    /**
+     * Copy properties from jqXhrToCopy to fakeJqXhr. This is makes fakeJqXhr
+     * behave properly.
+     */
+    function fakeJqXhrInheritance(fakeJqXhr, jqXhrToCopy) {
+        fakeJqXhr.readyState = jqXhrToCopy.readyState;
+        fakeJqXhr.status = jqXhrToCopy.status;
+        fakeJqXhr.statusText = jqXhrToCopy.statusText;
+        fakeJqXhr.responseXML = jqXhrToCopy.responseXML;
+        fakeJqXhr.responseText = jqXhrToCopy.responseText;
+        fakeJqXhr.responseJSON = jqXhrToCopy.responseJSON;
+        fakeJqXhr.getResponseHeader = jqXhrToCopy.getResponseHeader.bind(jqXhrToCopy);
+        fakeJqXhr.getAllResponseHeaders = jqXhrToCopy.getAllResponseHeaders.bind(jqXhrToCopy);
+        fakeJqXhr.setRequestHeader = jqXhrToCopy.setRequestHeader.bind(jqXhrToCopy);
+        fakeJqXhr.overrideMimeType = jqXhrToCopy.overrideMimeType.bind(jqXhrToCopy);
+        fakeJqXhr.statusCode = jqXhrToCopy.statusCode.bind(jqXhrToCopy);
+        fakeJqXhr.abort = jqXhrToCopy.abort.bind(jqXhrToCopy);
     }
 
-    return {
-        enable: function (newToken, newConfig) {
-            newConfig || (newConfig = {});
+    /**
+     * Patch $.ajax to support expired csrf tokens. If a request is made and the
+     * token is expired, then a new  token is fetched from the server. The original
+     * request will be run again with the new token.
+     *
+     * For the outside world only 1 request is send, but depending on the situation
+     * at most 3 request can be executed.
+     */
+    $.ajax = function (url, options) {
+        const pResult = $.Deferred(); // eslint-disable-line new-cap
+        const fakeJqXhr = pResult.promise();
 
-            if (!newToken) {
-                console.warn('Csrf token is not set!');
-            }
+        if (typeof url === 'object') {
+            options = url;
+            url = undefined;
+        } else {
+            options.url = url;
+        }
+
+        // The original ajax request might have success or error callbacks. We want
+        // to trigger them manually based on if there is a csrf token mismatch.
+        const success = options.success;
+        const error = options.error;
+        delete options.success;
+        delete options.error;
+
+        // Fire the first try!
+        const xhrFirstTry = originalAjax(options);
+
+        xhrFirstTry.error((jqXHR, textStatus, errorThrown) => {
+            if (jqXHR.status === 403) {
+                // We assume that a csrf token mismatch happend, so fetch a new
+                // token and retry with the correct token.
+                originalAjax(retryURL).done((data) => {
+                    setToken(parseResponse(data));
+                    let xhrSecondTry = null;
+
+                    options.success = (dataSecondSuccess, textStatusSecondSuccess, jqXHRSecondSuccess) => {
```

Comment:
compared to the original, I remove the duplicate `fakeJqXhrInheritance(fakeJqXhr, xhrSecondTry);` line in each callback (.success and .error)